src/main/scala/uk/gov/hmrc/play/config/inject/ServicesConfig.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.play.config.inject
+
+import com.google.inject.Inject
+import play.api.{Configuration, Environment}
+
+import scala.concurrent.duration.Duration
+
+trait ServicesConfig extends RunMode {
```

Comment:
It seems odd that the ServicesConfig logic (setting names, overrides/cascades etc) is duplicated in an inject and a non-inject version of the trait, and it means there's a chance the two versions could diverge in future.

Can they share the logic somehow? Perhaps they can both extend the same trait, or perhaps they can even _be_ the same trait. As long as the mechanism for obtaining the `Configuration` and `Environment` isn't baked in I don't see why the same trait can't be used in both DI and non-DI situations.